### DISCUSSION TITLE
Product options only returned if published to Online Store channel

### DISCUSSION THREAD

**Discussion by corygibbons** [Shopify/storefront-api-feedback#251] [Help] [answered]
Not sure if I'm just missing some docs somewhere or if this is a bug, seems counterintuitive.

Queries for product options, either via the now deprecated `options.values` field or `options.optionValues` returns an empty array unless the product is published to the Online Store.

```graphql
query product {
  product(handle: "some-handle") {
    handle
    title
    options(first: 10) {
      name
      id

      # Either of these returns an empty array
      values
      optionValues {
        name
      }
    }
  }
}
```

**Comment by vixdug** [comment]
Hey @corygibbons agree that does not sound intentional. Let me flag with one of our internal teams and follow up

**Comment by vixdug** [comment, marked as answer]
....and its resolved already! Let me know if you are still having issues.
